### PR TITLE
Add Air Quality Tutorial

### PR DESCRIPTION
**Pull request by f-schi**
This tutorial adds an end-to-end example of 1D time-series forecasting, from data loading to training and evaluation.

It introduces three new components:
| Component | Description |
|-----------|-------------|
| `AirQuality` / `AirQualityDataModule` | Hourly air-quality dataset (UCI, 2004–2005) with chronological split and normalization |
| `LTAE` | Lightweight Temporal Attention Encoder mapping `(B, T, C)` sequences to `(B, d)` embeddings |
| `TemporalRegressionTask` | Lightning module combining encoder, regression head, loss, and metrics |

**Goal:** predict the next `NUM_FUTURE_STEPS` hours of all sensor channels from the previous `NUM_PAST_STEPS` hours.

## AI Usage Disclosure

- [ ] 🟢 **No AI usage**: written by humans, for humans
- [x] 🟡 **AI-assisted**: AI helped with the coding, but I understand every line
- [ ] 🔴 **AI-generated**: AI did everything, review with caution

### CODE REVIEW

**Comment by f-schi** [comment]
While integrating `LTAE`, I had to apply a temporary monkey patch to disable `inplace` behavior in `nn.Dropout`. This was necessary because the current implementation of `LTAE` is incompatible with inplace dropout operations in the training setup.

```python
# Temporary workaround: disable inplace dropout in LTAE
_original_init = LTAE.__init__

def patched_init(self, *args, **kwargs):
    _original_init(self, *args, **kwargs)
    for m in self.modules():
        if isinstance(m, nn.Dropout):
            m.inplace = False

LTAE.__init__ = patched_init

**Comment by f-schi** [comment]
The current results indicate that the model performance is mediocre.
This likely needs to be investigated across multiple levels.
See the linked plots for details on the observed behavior and failure modes.
Screenshot 2026-05-07 105936
Screenshot 2026-05-07 105948
Screenshot 2026-05-07 110002
Air_2

**Comment by adamjstewart** [comment]
Can you add the AI usage disclosure from the PR template? This helps me prioritize which PRs to review and which to ignore.

**Comment by adamjstewart** [comment]
Let me know when you've incorporated all of the changes from the reviewers and the tests pass, then I'll review again.
